Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -759,6 +768,12 @@ extension SQLiteBookmarks {
                 return deferMaybe(result)
         }
     }
+
+    public func deleteLocalBookmarks(_ guids: [GUID]) -> Success {
```

Comment:
This shouldn't be here — it should be in `SQLiteBookmarksSyncing.swift` — and it shouldn't use the model, which also means you can get rid of the changes to `CachingItemSource` and `BookmarksModel`.

The model stuff sits between storage and the UI. Sync is on the other side: it talks directly to storage.

We don't currently allow users to bulk-delete in the UI; when we do, _then_ we'll want something like `deleteLocalBookmarks:guids`.

Consider also that model-like operations need to do sync-related metadata tracking to ensure that we upload deletions. You expressly want the opposite here: you're dropping records that we just uploaded.

You might consider producing a `LocalOverrideCompletionOp` from your upload, and then applying that with `applyLocalOverrideCompletionOp`. Set `processedLocalChanges` to the GUIDs that you uploaded. That method tackles SQL variable batching etc. for you.